express-notes-api/index.js

```diff
@@ -0,0 +1,109 @@
+const express = require('express');
+const data = require('./data.json');
+const fs = require('fs');
+const app = express();
+
+app.listen(3000, () => {
+  // eslint-disable-next-line no-console
+  console.log('The express server is listening on port 3000!');
+});
+
+app.use(express.json());
+
+app.get('/api/notes', function (req, res, next) {
+  res.status(200);
+  res.send(data);
+});
+
+app.get('/api/notes/:id', function (req, res, next) {
+  const paramID = parseInt(req.params.id);
+  if (typeof paramID !== 'number' || Math.sign(paramID) === -1) {
+    res.status(400);
+    res.send({ error: 'id is not a positive integer' });
+  } else {
+    if (data.notes[req.params.id]) {
+      res.status(200);
+      res.send(data.notes[req.params.id]);
+    } else {
+      res.status(404);
+      res.send({ error: 'Entry not found' });
+    }
+  }
+});
+
+app.post('/api/notes', function (req, res, next) {
+  if (JSON.stringify(req.body) !== '{}') {
```

Comment:
Here you want to specifically check if a `content` property exists on the body. This will just check if the body is empty or not but could have any property on it and it will go through. You would want to do something like:

```JAVASCRIPT
if (req.body.content) {
  // Add new note
} else {
  // Send error
}
```